packages/opencensus-instrumentation-http/src/http.ts

```diff
@@ -0,0 +1,303 @@
+/**
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {types} from '@opencensus/opencensus-core';
+import {classes} from '@opencensus/opencensus-core';
+import {logger} from '@opencensus/opencensus-core';
+import * as semver from 'semver';
+import * as shimmer from 'shimmer';
+import * as url from 'url';
+import * as uuid from 'uuid';
+
+// import * as Debug from 'debug';
+// const debug = Debug('opencensus');
+
+export class HttpPlugin extends classes.BasePlugin {
+  logger: types.Logger;
+
+  /** Constructs a new HttpPlugin instance. */
+  constructor(moduleName: string) {
+    super(moduleName);
+  }
+
+  /**
+   * Patches HTTP incoming and outcoming request functions.
+   * @param moduleExporters The HTTP package.
+   * @param tracer A tracer instance to create spans on.
+   * @param version The package version.
+   */
+  // tslint:disable:no-any
+  applyPatch(moduleExporters: any, tracer: types.Tracer, version: string) {
```

Comment:
> moduleExports rather than moduleExporters. 
Indeed I already know about this typo. But is a typo from BasePlugin from Core that has a property named moduleExporters, so we should keep this name until correct the typo from BasePlugin in a specific PR - or break this rule and rename the BasePlugin from Core right now. What do you think? I will add a TODO even so. 
  
> Also, the type here is typeof http (after import * as http from 'http'), is that correct?
Yes, it should be, but we were not importing http. Maybe we should. I will analyze that.